numba/np/arraymath.py

```diff
@@ -4836,3 +4842,162 @@ def impl(filt, trim='fb'):
 
     return impl
+
+
+@overload(np.setxor1d)
+def jit_np_setxor1d(ar1, ar2, assume_unique=False):
+    if not (type_can_asarray(ar1) or type_can_asarray(ar2)):
+        raise TypingError('setxor1d: first two args must be array-like')
+    if not (isinstance(assume_unique, (types.Boolean, bool))):
+        raise TypingError('setxor1d: Argument assume_unique must be boolean')
+
+    # https://github.com/numpy/numpy/blob/
+    # 03b62604eead0f7d279a5a4c094743eb29647368/numpy/lib/arraysetops.py#L477
+    def np_setxor1d_impl(ar1, ar2, assume_unique=False):
+        a = np.asarray(ar1)
+        b = np.asarray(ar2)
+
+        if not assume_unique:
+            a = np.unique(a)
+            b = np.unique(b)
+        else:
+            a = a.ravel()
+            b = b.ravel()
+
+        # Implementation very similar to np_intersect1d_impl:
+        # We want union minus the intersect
+        aux = np.concatenate((a, b))
+        aux.sort()
+
+        flag = np.empty(aux.shape[0] + 1, dtype=np.bool_)
+        flag[0] = True
+        flag[-1] = True
+        flag[1:-1] = aux[1:] != aux[:-1]
+        return aux[flag[1:] & flag[:-1]]
+
+    return np_setxor1d_impl
+
+
+@overload(np.setdiff1d)
+def jit_np_setdiff1d(ar1, ar2, assume_unique=False):
+    if not (type_can_asarray(ar1) or type_can_asarray(ar2)):
+        raise TypingError('setdiff1d: first two args must be array-like')
+    if not (isinstance(assume_unique, (types.Boolean, bool))):
+        raise TypingError('setdiff1d: Argument assume_unique must be boolean')
+
+    # https://github.com/numpy/numpy/blob/
+    # 03b62604eead0f7d279a5a4c094743eb29647368/numpy/lib/arraysetops.py#L940
+    def np_setdiff1d_impl(ar1, ar2, assume_unique=False):
+        ar1 = np.asarray(ar1)
+        ar2 = np.asarray(ar2)
+        if assume_unique:
+            ar1 = ar1.ravel()
+            ar2 = ar2.ravel()
+        else:
+            ar1 = np.unique(ar1)
+            ar2 = np.unique(ar2)
+        return ar1[np.in1d(ar1, ar2, assume_unique=True, invert=True)]
+
+    return np_setdiff1d_impl
+
+
+@overload(np.in1d)
+def jit_np_in1d(ar1, ar2, assume_unique=False, invert=False, kind="sort"):
+    if not (type_can_asarray(ar1) or type_can_asarray(ar2)):
+        raise TypingError('in1d: first two args must be array-like')
+    if not (isinstance(assume_unique, (types.Boolean, bool))):
+        raise TypingError('in1d: Argument "assume_unique" must be boolean')
+    if not (isinstance(invert, (types.Boolean, bool))):
+        raise TypingError('in1d: Argument "invert" must be boolean')
+    if not (isinstance(kind, (types.UnicodeType, str))):
+        raise TypingError('in1d: Argument "kind" must be string')
+
+    # https://github.com/numpy/numpy/blob/
+    # 03b62604eead0f7d279a5a4c094743eb29647368/numpy/lib/arraysetops.py#L525
+    def np_in1d_impl(ar1, ar2, assume_unique=False, invert=False, kind="sort"):
+        if kind != "sort":
+            raise ValueError('Only kind="sort" is supported')
```

Comment:
Since `kind="sort"` is the only accepted value for this keyword argument, move the check to be a compile-time one, rather than an exception raised at runtime.